equinox/nn/_weight_norm.py

```diff
@@ -0,0 +1,100 @@
+from typing import Generic, Optional, TypeVar, Union
+
+import jax
+import jax.numpy as jnp
+from jaxtyping import Array, PRNGKeyArray
+
+from .._module import field, Module
+from .._tree import tree_at
+
+
+_Layer = TypeVar("_Layer")
+
+
+def _norm_except_axis(
+    v: Array, pow: Optional[Union[int, str]] = None, axis: Optional[int] = 0
+) -> Array:
+    for ax in range(len(v.shape)):
+        if ax != axis:
+            v = jnp.linalg.norm(v, ord=pow, axis=ax, keepdims=True)
+    return v if axis is not None else v.reshape([])
```

Comment:
I think it should be possible to express this function as just a single `norm`, without needing to make repeated calls. (And if `axis is not None` then `jax.vmap` over that axis.)